lib/Serialization/ScanningLoaders.cpp

```diff
@@ -83,6 +83,30 @@ std::error_code SwiftModuleScanner::findModuleFilesInDirectory(
   return dependencyInfo.getError();
 }
 
+bool SwiftModuleScanner::canImportModule(
+    ImportPath::Module path, SourceLoc loc, ModuleVersionInfo *versionInfo,
+    bool isTestableDependencyLookup) {
+  if (path.hasSubmodule())
+    return false;
+
+  // Check explicitly-provided Swift modules with '-swift-module-file'
```

Comment:
Did this code path ignores the versionInfo? I think we still cares about version when doing a `#canImport` check on explicitly provided module. Will be good to see a test case for this too.

Reply:
This code path ignores version info, yeah. 
`#if canImport` is handled earlier in the scanner, and we actually already instantiate an `ExplicitSwiftModuleLoader` for this purpose which handles it accordingly. Which means that before this change things were a bit broken because the scanner *did* check `canImport` statements against `-swift-module-file` inputs but then did not actually use them to resolve dependencies. 

Added a test for this.